core/src/main/kotlin/fr/sncf/osrd/api/api_v2/standalone_sim/SimulationEndpoint.kt

```diff
@@ -51,13 +52,20 @@ class SimulationEndpoint(
             val chunkPath = makeChunkPath(infra.rawInfra, request.path.trackSectionRanges)
             val routePath = convertRoutePath(infra.rawInfra, request.path.routes)
             val pathProps = makePathProperties(infra.rawInfra, chunkPath, routePath.toList())
+            val blockPath = mutableStaticIdxArrayListOf<Block>()
+            for (blockName in request.path.blocks) {
+                val blockId = infra.blockInfra.getBlockFromName(blockName)
+                assert(blockId != null) { "An input block from path is missing in infrastructure" }
+                blockPath.add(blockId!!)
```

Comment:
*Ideally* we'd create an error similar to `UnknownRoute`, and we could write it like this:

```suggestion
                val blockId = infra.blockInfra.getBlockFromName(blockName)
                    ?: throw OSRDError(ErrorType.UnknownBlock)
                blockPath.add(blockId)
```

Reply:
Done in https://github.com/OpenRailAssociation/osrd/pull/10147/commits/ce97208b8579677e096b53dad101cada8b79a540